source/extensions/common/wasm/wasm.cc

```diff
@@ -462,6 +467,22 @@ Word setMetadataStructHandler(void* raw_context, Word type, Word name_ptr, Word
                                                                name.value(), value.value())));
 }
 
+// Generic selector
+Word getSelectorExpressionHandler(void* raw_context, Word path_ptr, Word path_size,
+                                  Word value_ptr_ptr, Word value_size_ptr) {
+  auto context = WASM_CONTEXT(raw_context);
+  auto path = context->wasmVm()->getMemory(path_ptr, path_size);
+  if (!path.has_value()) {
+    return wasmResultToWord(WasmResult::InvalidMemoryAccess);
+  }
+  auto value = context->getSelectorExpression(path.value());
+  if (value.has_value()) {
+    context->wasm()->copyToPointerSize(value.value(), value_ptr_ptr, value_size_ptr);
```

Comment:
You need to handle the return value here and return a memory error on false.

Reply:
Thanks, done.